packages/react/src/hooks/network-status/useBlockNumber.ts

```diff
@@ -41,19 +42,22 @@ export function useBlockNumber({
   onSuccess,
 }: UseBlockNumberArgs & UseBlockNumberConfig = {}) {
   const chainId = useChainId({ chainId: chainId_ })
-  const provider = useProvider()
-  const webSocketProvider = useWebSocketProvider()
+  const provider = useProvider({ chainId })
+  const webSocketProvider = useWebSocketProvider({ chainId })
   const queryClient = useQueryClient()
 
   React.useEffect(() => {
     if (!watch && !onBlock) return
 
-    const listener = (blockNumber: number) => {
+    // We need to debounce the listener as we want to opt-out
+    // of the behavior where ethers emits a "block" event for
+    // every block that was missed in between the `pollingInterval`.
+    const listener = debounce((blockNumber: number) => {
       // Just to be safe in case the provider implementation
       // calls the event callback after .off() has been called
       if (watch) queryClient.setQueryData(queryKey({ chainId }), blockNumber)
       if (onBlock) onBlock(blockNumber)
-    }
+    }, 50)
```

Comment:
Why `50`?

Reply:
We can't really determine a static time to wait without knowing the average block interval for the chain. For chains that increment the block number for every transaction, this could get very costly, so it seems that 50 is on the safe side. Open to other ideas too.